### PR TITLE
feat(cherry-pick-bot): add support for preserving original PR title

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds support to preserve original PR title to cherry-pick bot.

### CODE REVIEW

**Comment by eddycharly** [comment]
cc @chingor13 

Would you be kind enough to help me on this ?

**Comment by tmatsuo** [comment]
Hi @eddycharly

First of all, thanks for your contribution!

My recommendations:

1. Add a test case to test/bot.ts
2. Add a config schema, schema check, and some tests for it

For the config schema, there are some examples in other bots (e.g. packages/snippet-bot) by utilizing the `bot-config-utils` library. Your bot can check the config file against the predefined schema (e.g. packages/snippet-bot/src/config-schema.json) and if someone adds a broken config file, it will post failing status check to that PR.

**Comment by eddycharly** [comment]
Hi @tmatsuo !

Thanks, I'm going to add a test and look at the config schema.

**Comment by eddycharly** [comment]
@tmatsuo does this look like a valid `config-schema.json` file ? Do I need to specify the properties are optional ?

**Comment by chingor13** [comment]
> Do I need to specify the properties are optional ?

JSONSchema properties should be optional by default (There's a `required` field to specify which fields are required if there are any).

File: packages/cherry-pick-bot/src/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/src/bot.ts:267]: This doesn't match the conventional commit.

File: packages/cherry-pick-bot/src/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/src/bot.ts:151]: This doesn't match conventional commit message.

How about to preserve the original title and add an indicator for being cherry-pick in the pull request body?

**Comment by eddycharly** [comment]
@tmatsuo @chingor13 I added the `ConfigChecker` support and tests. PTAL

**Review by tmatsuo — COMMENTED** [review]
@eddycharly 

I added some comments. I also have some asks:

1. Can you add the code like the following?

```
import schema from './config-schema.json';

// Then add the schema to `getConfig`
    const remoteConfig = await getConfig<Configuration>(
      octokit,
      owner,
      repo,
      CONFIGURATION_FILE_PATH,
      {schema: schema}
    );
```

This will ensure the returned object matches the schema.

2. Revert changes in package-lock.json

We want to install new dependencies with certain stabilityDays.

3. Consider adding a new handler for validating the config schema when someone tries to add the config file? You need to add a new handler like the following:

```
  app.on(
    [
      'pull_request.opened',
      'pull_request.reopened',
      'pull_request.edited',
      'pull_request.synchronize',
    ],
```

Add some boilerplate code for getting logger and octokit then add the following:

```
      await configChecker.validateConfigChanges(
        octokit,
        owner,
        repo,
        context.payload.pull_request.head.sha,
        context.payload.pull_request.number
      );
```

Thanks

**Comment by tmatsuo** [comment]
@eddycharly 

and yes, please also update the README

File: packages/cherry-pick-bot/src/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/src/bot.ts:267]: Would adding `cherry-pick` at the end be ok ?

Something like `${context.payload.pull_request.title} (cherry-pick #${context.payload.pull_request.number})`.

File: packages/cherry-pick-bot/src/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/src/bot.ts:169]: This handler is only called when a pull request is closed.

We need to add a new handler when a pull request is opened/synchronized/reopened.

File: packages/cherry-pick-bot/src/bot.ts
Comment (tmatsuo) [reply packages/cherry-pick-bot/src/bot.ts:267]: Yeah looks good

File: packages/cherry-pick-bot/src/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/src/bot.ts:169]: Added.

File: packages/cherry-pick-bot/src/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/src/bot.ts]: Fixed

File: packages/cherry-pick-bot/src/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/src/bot.ts]: Fixed

**Comment by eddycharly** [comment]
@tmatsuo I think i addressed your comments and and requested changes.

I also updated the `README`.

Thanks for your help so far :pray:

File: packages/cherry-pick-bot/src/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/src/bot.ts:300]: This happens against a pull request. If we want to log this, maybe add pr_number too.

File: packages/cherry-pick-bot/test/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/test/bot.ts:66]: is `getConfigStub` needed?

File: packages/cherry-pick-bot/test/bot.ts
Comment (tmatsuo) [inline packages/cherry-pick-bot/test/bot.ts:175]: Is this stub required?

**Review by tmatsuo — COMMENTED** [review]
Looking good with possible clean ups.

File: packages/cherry-pick-bot/test/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/test/bot.ts]: Not anymore, removed

File: packages/cherry-pick-bot/test/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/test/bot.ts]: Good catch, removed

File: packages/cherry-pick-bot/src/bot.ts
Comment (eddycharly) [reply packages/cherry-pick-bot/src/bot.ts]: Added `pr_number` to the log.

**Comment by eddycharly** [comment]
@tmatsuo thanks for your suggestions. I made the changes, PTAL.

**Review by tmatsuo — COMMENTED** [review]
LGTM, but we'd want one more LGTM before merging.

@chingor13 Can you also take a look?

**Comment by eddycharly** [comment]
Thanks @tmatsuo I hope I covered all suggestions and tests :crossed_fingers:

**Review by chingor13 — APPROVED** [review]

**Review by tmatsuo — APPROVED** [review]

**Comment by tmatsuo** [comment]
@eddycharly It's deployed. Thanks for your contribution!

**Comment by eddycharly** [comment]
@tmatsuo thank you so much for helping me on it !

**Pull request merged**
